### PR TITLE
Initial commit; Document initial model of the solution

### PR DESCRIPTION
**Pull request by siripr4**

### CODE REVIEW

**Comment by shhivam** [comment]
I don't have any other comments yet other than what we [already discussed](https://nilenso.slack.com/archives/C044TS6RY3C/p1668427611204799). 

I do have some comments on the commit:
1. The one commit contains 15 file changes. Ideally, a commit should never be this large. Think about how you can break down the commit even further and make them atomic.
    - Although there aren't many commits, consider using git's interactive rebase to do this. This will help you get familiar with git rebase as well if you are not.
3. I think it will be safe to add the whole `.idea` folder in `.gitignore`

Also, I work at odd hours as per my convenience, so please don't take the timestamp of this message as representative of working late 😁

**Comment by yogirajh007** [comment]
+1 to what @shhivam said about 15 changed files. Ideally, most of them should be in .gitignore. 

Also, take a look at https://www.conventionalcommits.org if not already. This has helped me a lot in making my commits better.

Went through the README, the approach for solving the problem seems fine. Looking forward to the code 😁

**Comment by siripr4** [comment]
@shhivam @yogirajh007 I've split the original commit to 3, breaking them down to 
1. Project creation
2. Documentation
3. Cleanup IDE generated files

The 1st commit still has 9 files (generated by `leiningen`). I think having many files during project creation is inevitable.
Let me know if there's a better way to do this perhaps

**Comment by siripr4** [comment]
> Also, take a look at https://www.conventionalcommits.org if not already. This has helped me a lot in making my commits better.

@yogirajh007 This was helpful, thanks! 🙂 

I was earlier using guidelines mentioned [here](https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53). However Conventional Commits does this more structured.

File: dice-engine/dice-engine.iml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (shhivam) [inline dice-engine/dice-engine.iml:1]: Is this file still needed?

Comment (siripr4) [reply dice-engine/dice-engine.iml]: Missed this. 
It's not needed. I've removed the file and updated .gitignore

**Comment by shhivam** [comment]
heh, thanks. Approving this PR.
Ideally, there should not be another commit fixing PR comments (except when the PR comment suggests a big change imo).

The commit [chore: add .iml files to .gitignore](https://github.com/nilenso/siris-onboarding/pull/2/commits/4dc77741fc64a3d79502aff8b908b79294994171) & [chore: remove dice-engine.iml](https://github.com/nilenso/siris-onboarding/pull/2/commits/4dbce2ad120a6c6147803c686d36dc7a2d0e071f) could be part of [chore: remove .idea directory from tracking](https://github.com/nilenso/siris-onboarding/pull/2/commits/88bdfe542e5d5f9d7adbb68f9bd903deea294189) only but I think we can let it be for now.

**Review by shhivam — APPROVED** [review]
